pjsip/src/pjsip/sip_transport.c

```diff
@@ -2156,41 +2156,70 @@ PJ_DEF(pj_ssize_t) pjsip_tpmgr_receive_packet( pjsip_tpmgr *mgr,
 
             msg_status = pjsip_find_msg(current_pkt, remaining_len, PJ_FALSE, 
                                         &msg_fragment_size);
-            if (msg_status != PJ_SUCCESS) {
-                if (remaining_len == PJSIP_MAX_PKT_LEN) {
-                    mgr->on_rx_msg(mgr->endpt, PJSIP_ERXOVERFLOW, rdata);
-                    
-                    /* Notify application about the message overflow */
-                    if (mgr->tp_drop_data_cb) {
-                        pjsip_tp_dropped_data dd;
-                        pj_bzero(&dd, sizeof(dd));
-                        dd.tp = tr;
-                        dd.data = current_pkt;
-                        dd.len = msg_fragment_size;
-                        dd.status = PJSIP_ERXOVERFLOW;
-                        (*mgr->tp_drop_data_cb)(&dd);
-                    }
-
-                    if (rdata->tp_info.transport->idle_timer.id == 
-                                                         INITIAL_IDLE_TIMER_ID)
-                    {
-                        /* We are not getting the first valid SIP message 
-                         * as expected, close the transport.
-                         */
-                        PJ_LOG(4, (THIS_FILE, "Unexpected data was received "\
-                            "while waiting for a valid initial SIP messages. "\
-                            "Shutting down transport %s",
-                            rdata->tp_info.transport->obj_name));
-
-                        pjsip_transport_shutdown(rdata->tp_info.transport);
```

Comment:
@trengginas I added the transport_shutdown in the new extended code path because I saw in the other code path it was already being done. So not sure if then this path was wrong then?
Let me know whether you want to merge as it, drop the pjsip_transport_shutdown() from the new path or from both paths.